src/functional/iterator/pretty_printer.py

```diff
@@ -0,0 +1,277 @@
+            vtup = self._vmerge(["{"], self._indent(args), ["}"])
+            return self._optimum(htup, vtup)
+
+        hfun = self._hmerge(fun, ["("], args, [")"])
+        vfun = self._vmerge(self._hmerge(fun, ["("]), self._indent(args), [")"])
+        return self._prec_parens(self._optimum(hfun, vfun), prec, PRECEDENCE["__call__"])
+
+    def visit_FunctionDefinition(self, node: ir.FunctionDefinition, prec: int) -> list[str]:
+        assert prec == 0
+        params = self.visit(node.params, prec=0)
+        expr = self.visit(node.expr, prec=0)
+
+        start, bridge = [node.id + " = λ("], [") → "]
+        if not params:
+            params = self._hmerge(start, bridge)
+        else:
+            hparams = self._hmerge(start, *self._hinterleave(params, ", "), bridge)
+            vparams = self._vmerge(start, *self._hinterleave(params, ",", indent=True), bridge)
+            params = self._optimum(hparams, vparams)
+
+        hbody = self._hmerge(params, expr)
+        vbody = self._vmerge(params, self._indent(expr))
+        return self._optimum(hbody, vbody)
+
+    def visit_StencilClosure(self, node: ir.StencilClosure, *, prec: int) -> list[str]:
+        assert prec == 0
+        domain = self.visit(node.domain, prec=0)
+        stencil = self.visit(node.stencil, prec=0)
+        output = self.visit(node.output, prec=0)
+        inputs = self.visit(node.inputs, prec=0)
+
+        hinputs = self._hmerge(["("], *self._hinterleave(inputs, ", "), [")"])
+        vinputs = self._vmerge(["("], *self._hinterleave(inputs, ",", indent=True), [")"])
+        inputs = self._optimum(hinputs, vinputs)
+
+        head = self._hmerge(output, [" ← "])
+        foot = self._hmerge(inputs, [" @ "], domain)
+
+        h = self._hmerge(head, ["("], stencil, [")"], foot)
+        v = self._vmerge(
+            self._hmerge(head, ["("]),
+            self._indent(self._indent(stencil)),
+            self._indent(self._hmerge([")"], foot)),
+        )
+        return self._optimum(h, v)
+
+    def visit_FencilDefinition(self, node: ir.FencilDefinition, *, prec: int) -> list[str]:
+        assert prec == 0
+        function_definitions = self.visit(node.function_definitions, prec=0)
+        closures = self.visit(node.closures, prec=0)
+        params = self.visit(node.params, prec=0)
+
+        hparams = self._hmerge([node.id + "("], *self._hinterleave(params, ", "), [") {"])
+        vparams = self._vmerge(
+            [node.id + "("], *self._hinterleave(params, ",", indent=True), [") {"]
+        )
+        params = self._optimum(hparams, vparams)
+
+        function_definitions = self._vmerge(*function_definitions)
+        closures = self._vmerge(*closures)
+
+        return self._vmerge(
+            params, self._indent(function_definitions), self._indent(closures), ["}"]
+        )
+
+    @classmethod
+    def apply(cls, node: ir.Node, indent: int = DEFAULT_INDENT, width: int = DEFAULT_WIDTH) -> str:
```

Comment:
If `pformat` is the preferred public API for the `PrettyPrinter`, I suggest to set the arguments' default values only there.

```suggestion
    def apply(cls, node: ir.Node, indent: int, width: int) -> str:
```